### PR TITLE
my solution

### PR DESCRIPTION
**Pull request by Pavlo000**

### CODE REVIEW

File: src/arrayMethodSort.js [hunk 1]
```diff
@@ -4,9 +4,37 @@
...
-    // write code here
+  [].__proto__.sort2 = function(compareFunction = defaultFunction) {
+    let permutations;
+
+    do {
+      permutations = false;
+
+      for (let i = 1; i < this.length; i++) {
+        if (compareFunction(this[i - 1], this[i]) > 0) {
+          [this[i - 1], this[i]] = [this[i], this[i - 1]];
```
Comment (stanislav-topikha) [inline src/arrayMethodSort.js:15]: 🔥

File: src/arrayMethodSort.js [hunk 2]
```diff
@@ -4,9 +4,37 @@
...
+          permutations = true;
+        }
+      }
+    } while (permutations);
+
+    return this;
   };
+
+  function defaultFunction(a, b) {
```
Comment (stanislav-topikha) [inline src/arrayMethodSort.js:24]: Unclear name for function, better change it

File: src/arrayMethodSort.test.js [hunk 3]
```diff
@@ -53,7 +53,7 @@ test('custom compare function, numbers: [3, 12, 2, 11].sort2((a, b) => a - b)',
   });
 
 test(
-  `custom compare function, strings: 
+  `custom compare function, strings:
```
Comment (stanislav-topikha) [inline src/arrayMethodSort.test.js:56]: No need to touch it at all)

**Review by stanislav-topikha — COMMENTED** [review]
👍

**Review by stanislav-topikha — APPROVED** [review]

**Review by mkyloa — APPROVED** [review]

**Pull request closed without merging**